### PR TITLE
chore: Added concepts for design tokens and web components.

### PR DESCRIPTION
**Pull request by tomheller**
### <strong>Pull Request</strong>
This pull request should be considered the start of the discussion about the introduction of design-tokens and web-components into barista. Please feel free to comment, contribute and discuss here. 
This is the beginning of this conversation.

### CODE REVIEW

**Comment by ffriedl89** [comment]
Really like the summary so far :) Very nice! 🚀 🎨 
No objections whatsoever, I think these constraints should help us ship more consistent components. The layer to control each component individually is awesome, this way we can swap out a spacing value on the button only if we notice that the button spacing is off.

**Comment by tomheller** [comment]
We should definitely consider some sort of visual impact or visual regression comparison for token changes. I have no idea what this could look like, but I assume this could be very helpful to see the impact of a change in a global fashion.

**Comment by ffriedl89** [comment]
I've added the exact same idea to my list of unordered ideas in notion :D Have a tool that shows the impact of the change in the design tokens. List of affected components, maybe with a before and after view :D  ¯\\_(ツ)_/¯

**Comment by thomaspink** [comment]
As far as i have understood design tokens (in combination with css variables), we also need to prefix all our css-variables because the cascade also applies to them.

File: concepts/design-tokens/README.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+logic. For some of these usecases, exported typescript constants for the design
+tokens could help us keep things in sync between typescript and style.
+
+Potential output could look like this:
+
+```ts
+/** ts-constants/button.ts*/
+export const baseButtonBackgroundColor = '#123456';
+export const baseButtonBackgroundColorHover = '#123456';
+export const primaryPuttonBackgroundColor = '#123456';
```
Comment (lara-aigmueller) [inline concepts/design-tokens/README.md]:
```suggestion
export const primaryButtonBackgroundColor = '#123456';
```

File: concepts/design-tokens/README.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+## Structure
+
+Tokens are usually stored in some sort of structual data (yml, json,
+database,...). Which format we are ultimately choosing is still up for
+discussion.
+
+Just like the format, the file structure is also up for discussion. There are
+multiple ways of structuring the design tokens into a folder structure, and a
+convertion tool like [theo] will let you customize
```
Comment (NinaKammerlander) [inline concepts/design-tokens/README.md]: conversion

File: concepts/design-tokens/README.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+> color, and spacing so that your design system can be shared across platforms
+> like iOS, Android, and regular ol' websites.
+
+-- <cite>Robin Rendle "[What are design tokens]", CSS-Tricks</cite>
+
+## What we want to achived with Design tokens
+
+In the current state of the barista components library, we do share some basic
+color, typography and spacing definitions through scss mixins and variables
+within the angular eco system. Which works very well for color definitions, did
```
Comment (NinaKammerlander) [inline concepts/design-tokens/README.md]: ecosystem

File: concepts/design-tokens/README.md (hunk 1 above)
Comment (tomheller) [reply concepts/design-tokens/README.md]: but my `primaryPutton` :(

**Comment by tomheller** [comment]
Thank you for spotting all my typos, they are resolved now.

**Comment by tomheller** [comment]
> I'm wondering if it makes sense to scope some CSS custom properties and not making them all available at root-level. But I'm sure this depends on when the consumer should be able to override those values and where and why.

I believe that this makes a lot of sense to scope the pattern/component tokens to their respective component. But we will need to figure out how we can or want to achieve that.

Possible downside would be, if you would want to restyle your buttons, you would have to restyle all of them over and over again vs. redefining the new variable definitions in a wrapper and let the cascade handle the rest (could be very helpful for theming if we would need that). Both options seem valid, I think this is something we need to discuss further. Any input is very much appreciated here.

**Comment by thomaspink** [comment]
Great work @tomheller !!

Just another very small input: 
Currently top level constants in typescript are (mostly - we forgot it sometimes) in all uppercase, with underscores between. Should we stick to this or switch to camelCase.

`export const dtBaseButtonBackgroundColor = '#123456';` 
vs 
`export const DT_BASE_BUTTON_BACKGROUND_COLOR = '#123456'`;

For consistency with the angular codebase i'd suggest to stick with the uppercase variant (as it is also more common in the TS community). But i also have no objections if we wanna change that. We should just agree on one way or the other for the whole codebase.

**Comment by tomheller** [comment]
>Currently top level constants in typescript are (mostly - we forgot it sometimes) in all uppercase, with underscores between. Should we stick to this or switch to camelCase.

Good point, I have updated the typescript tokens section accordingly

File: concepts/design-tokens/README.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+discussion.
+
+Just like the format, the file structure is also up for discussion. There are
+multiple ways of structuring the design tokens into a folder structure, and a
+conversion tool like [theo] will let you customize
+[input and output formats quite easily](#output-and-formats).
+
+Here are some variations of design tokens structure in the wild:
+
+- [Salesforce design tokens]
```
Comment (lukasholzer) [inline concepts/design-tokens/README.md]: should this be a link?

Comment (tomheller) [reply concepts/design-tokens/README.md]: Yes, and it is. This is a shorthand link syntax in markdown. 
Link targets are collected at the bottom of the md file: 
https://github.com/dynatrace-oss/barista/pull/633/files#diff-3f1f6d3a5d17588e9f3aa79c5b52262eR228

**Comment by ffriedl89** [comment]
I think this is great content, should we merge it? And where do we wanna have the concept section popup in barista? Should we add an internal next section on top level?

**Comment by tomheller** [comment]
> I think this is great content, should we merge it? And where do we wanna have the concept section popup in barista? Should we add an internal next section on top level?

I don't know? Does this make sense to include in the published barista? It's more about specs / guidelines within the component library.

**Comment by tomheller** [comment]
Needed to reword the commit, because CI was not running on it. Review has been requested again.

**Pull request merged**